packages/nodes-base/nodes/MessageBird/MessageBird.node.ts

```diff
@@ -0,0 +1,354 @@
+					},
+					{
+						displayName: 'Reference',
+						name: 'reference',
+						type: 'string',
+						default: '',
+						description: 'A client reference.'
+					},
+					{
+						displayName: 'Report Url',
+						name: 'reportUrl',
+						type: 'string',
+						default: '',
+						description:
+							'The status report URL to be used on a per-message basis.<br /> Reference is required for a status report webhook to be sent.'
+					},
+					//date-time format
+					{
+						displayName: 'Scheduled Date-time',
+						name: 'scheduledDatetime',
+						type: 'dateTime',
+						default: '',
+						placeholder: '2011-08-30T09:30:16.768-04:00',
+						description:
+							'The scheduled date and time of the message in RFC3339 format (Y-m-dTH:i:sP).'
+					},
+					{
+						displayName: 'Type',
+						name: 'type',
+						type: 'options',
+						options: [
+							{
+								name: 'sms',
+								value: 'sms'
+							},
+							{
+								name: 'binary',
+								value: 'binary'
+							},
+							{
+								name: 'flash',
+								value: 'flash'
+							}
+						],
+						default: '',
+						description:
+							'The type of message.<br /> Values can be: sms, binary, or flash.'
+					},
+					//hash
+					{
+						displayName: 'Type Details',
+						name: 'typeDetails',
+						type: 'string',
+						default: '',
+						description:
+							'A hash with extra information.<br /> Is only used when a binary message is sent.'
+					},
+					{
+						displayName: 'Validity',
+						name: 'validity',
+						type: 'number',
+						default: '',
+						typeOptions: {
+							minValue: 1
+						},
+						description: 'The amount of seconds that the message is valid.'
+					}
+				]
+			}
+		]
+	};
+
+	async execute(this: IExecuteFunctions): Promise<INodeExecutionData[][]> {
+		const items = this.getInputData();
+		const returnData: IDataObject[] = [];
+
+		let operation: string;
+		let resource: string;
+
+		// For POST
+		let bodyRequest: IDataObject;
+		// For Query string
+		let qs: IDataObject;
+
+		let requestMethod;
+
+		for (let i = 0; i < items.length; i++) {
+			qs = {};
+
+			resource = this.getNodeParameter('resource', i) as string;
+			operation = this.getNodeParameter('operation', i) as string;
+
+			if (resource === 'sms') {
+				//https://developers.messagebird.com/api/sms-messaging/#sms-api
+				if (operation === 'send') {
+					// ----------------------------------
+					//         sms:send
+					// ----------------------------------
+
+					requestMethod = 'POST';
+					const originator = this.getNodeParameter('originator', i) as string;
+					const body = this.getNodeParameter('message', i) as string;
+
+					bodyRequest = {
+						recipients: [],
+						originator,
+						body
+					};
+					const additionalFields = this.getNodeParameter(
+						'additionalFields',
+						i
+					) as IDataObject;
+
+					if (additionalFields.groupIds) {
+						bodyRequest.groupIds = additionalFields.groupIds as string;
+					}
```

Comment:
Might be able to compress all these like:

```js
additionalFields.groupIds ? bodyRequest.groupIds = additionalFields.groupIds as string;
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator